libs/platform/src/lib/components/form/text-area/text-area.component.ts

```diff
@@ -0,0 +1,402 @@
+        // set max height
+        this._setMaxHeight();
+
+        // don't set any error state if we are not showing counter message
+        if (!this.showExceededText) {
+            this.stateType = null;
+        }
+
+        super.ngAfterViewInit();
+    }
+
+    /** @hidden */
+    ngAfterViewChecked() {
+        // when value is custom set(initial value is present), the heights don't get
+        // set to show the full text in the ngAfterViewInit immediately. therefore, we call autoGrowTextArea in
+        // ngAfterViewChecked and detect changes again.
+        if (this.value && this.isValueCustomSet) {
+            this.autoGrowTextArea(); // if applicable, grow the textarea when value is custom set
+            this.stateChanges.next('textarea: ngAfterViewChecked');
+        }
+    }
+
+    /** write value for ControlValueAccessor */
+    writeValue(value: any): void {
+        if (value) {
+            super.writeValue(value);
+            this.updateCounterInteractions();
+            this.stateChanges.next('textarea: writeValue');
+        }
+    }
+
+    /** update the counter message and related interactions */
+    updateCounterInteractions(): void {
+        if (this.value) {
+            this._textAreaCharCount = this.value.length;
+            if (this.maxLength) {
+                // newly added to avoid unnecessary iteration, remove if issue found
+                this.validateLengthOnCustomSet();
+            }
+        }
+    }
+
+    /** if exceeded maxlength when set as a value in code, highlight the exceeded text. */
+    validateLengthOnCustomSet(): void {
+        if (this._textAreaCharCount > this.maxLength) {
+            if (this._isPasted) {
+                this.textareaElement.nativeElement.focus();
+                // this.textareaElement.nativeElement.setSelectionRange(10, 20);
+                this.textareaElement.nativeElement.setSelectionRange(this.maxLength, this._textAreaCharCount);
+            }
+            this.counterExcessOrRemaining = this.excessText;
+            this.exceededCharCount = this._textAreaCharCount - this.maxLength;
+        } else {
+            this.counterExcessOrRemaining = this.remainingText;
+            this.exceededCharCount = this.maxLength - this._textAreaCharCount;
+        }
+        // reset flag
+        this._isPasted = false;
+    }
+
+    /** handle exceeded maxlength case when text is pasted */
+    handlePasteInteraction(): void {
+        this._isPasted = true;
+        /// For IE
+        if (window['clipboardData']) {
+            const value = window['clipboardData'].getData('Text');
+            // todo: handle for IE
+            this.updateCounterInteractions();
+        } else {
+            // for other navigators
+            navigator['clipboard'].readText().then((clipText) => {
+                this.updateCounterInteractions();
+            });
+        }
+    }
+
+    /** handle auto growing of textarea */
+    autoGrowTextArea() {
```

Comment:
Why don't you bring cdk auto resize?

Reply:
cdk's auto resizing works only on rows, and expects min rows and max rows to be given. Fiori requirement does not allow user to specify min rows. Fiori requirement also specifies growing by height, in addition to growing by rows(or maxLines), and when both are specified, height should be given preference.
Since some very specific requirements are present, we didn't use cdk features.